content/blog/2018-09-04-birds-taxo-traits.md

```diff
@@ -0,0 +1,559 @@
+itself with
+`ggtree`](https://guangchuangyu.github.io/2018/03/annotating-phylogenetic-tree-with-images-using-ggtree-and-ggimage/)
+but I’ll add them on the side instead.
+
+``` r
+# Get pics ids
+load(file.path("taxo", "ids.RData"))
+
+# Plot basic tree
+p <- ggtree::ggtree(tree$phylo) 
+
+# Sort the orders by node id
+orders <- dplyr::arrange(orders, - id)
+
+# Helper to plot one order
+plot_order <- function(order, orders,
+                       ids, p){
+                       
+  # Get index
+  i <- which(orders$order == order)
+  
+  # From image ID get image itself 
+  # and image metadata (copyright &co)
+  img_id <- ids$pic_id[ids$order == order]
+  img <- rphylopic::image_data(img_id, 512)
+  img_info <- rphylopic::image_get(img_id,
+                                   options = c("credit",
+                                               "licenseURL"))
+  
+  if(is.null(img_info$credit)){
+    img_info$credit <- ""
+  }
+  
+  # Now, plot!
+  p +
+  # Highlight the order
+  ggtree::geom_hilight(node = orders$id[i],
+                       fill = "salmon") +
+  # Order name as title
+  ggtitle(orders$order[i])+
+  xlim(0, 150) +
+  ylim(0, 250) +
+  # Add species names on the side
+  annotate("text", x = 110,
+           y = 200, label = orders$species[i],
+           size = 4) +
+  # Credit at the bottom
+  annotate("text", x = 110,
+           y = 0,
+           size = 2,
+          label = glue::glue("Silhouette: {img_info$credit}\n{img_info$licenseURL}"))
+  
+  # Save a first time
+  filepath <- file.path("taxo", glue::glue("p{i}.png"))
+  ggsave(filepath, width = 7, height = 7)
+  
+  # Add silhouette via magick
+  silhouette <- magick::image_read(img[[1]])
+  magick::image_read(filepath) %>%
+    magick::image_composite(silhouette,
+                            offset = "+1300+1400") %>%
+    magick::image_write(filepath)
+  
+}
+
+# Create aaall plots
+  
+purrr::walk(orders$order, plot_order,
+            orders, ids, p)
+
+```
+
+Once we have created all these PNGs, we can join them into a gif using
+[Jeroen Ooms’
+`gifski`](https://ropensci.org/technotes/2018/07/23/gifski-release/).
+
+``` r
+png_files <- fs::dir_ls("taxo", regexp = "[.]png$")
+
+gifski::gifski(png_files = png_files,
+               gif_file = file.path("2018-09-04-birds-taxo-traits_files",
+                                    "figure-markdown_github", "taxo.gif"),
+               delay = 3,
+               width = 500, height = 500,
+               progress = FALSE)
+```
+
+    ## [1] "/img/blog-images/2018-09-04-birds-taxo-traits/taxo.gif"
+
+``` r
+knitr::include_graphics(file.path("2018-09-04-birds-taxo-traits_files",
+                                    "figure-markdown_github", "taxo.gif")) 
+```
+
+![animated tree with species names and order
+silhouette](/img/blog-images/2018-09-04-birds-taxo-traits/taxo.gif)
+
+This gif shows many species names and order giving us a feeling for what
```

Comment:
"many species names and order"
can you clarify what you mean by "order" here? 

should "Order" be capitalized in post when referring to taxonomic classification to avoid confusion?
I see an example here "[Homo sapiens is the most abundant member of the Order Primates](https://sciencemadeeasy.kinja.com/how-to-correctly-write-about-taxonomic-nomenclature-1623703107)"

Reply:
Added a link to https://en.wikipedia.org/wiki/Category:Bird_orders but "order" is not capitalized there? I can still capitalize if you prefer though.